plugins/modules/digital_ocean_droplet.py

```diff
@@ -271,12 +273,25 @@ def get_droplet(self):
             json_data = self.get_by_name(self.module.params['name'])
         return json_data
 
-    def create(self):
+    def create(self, state):
         json_data = self.get_droplet()
         droplet_data = None
-        if json_data:
+        if json_data:  # The droplet exists already
             droplet_data = self.get_addresses(json_data)
-            self.module.exit_json(changed=False, data=droplet_data)
+            # If state is active or inactive, ensure requested and desired power states match
+            droplet = json_data.get('droplet', None)
+            if droplet is not None:
+                droplet_id = droplet.get('id', None)
+                droplet_status = droplet.get('status', None)
+                if droplet_id is not None and droplet_status is not None:
+                    if state == 'active' and droplet_status != 'active':
+                        json_data = self.ensure_power_on(droplet_id)
```

Comment:
This part was confusing to me because you're grabbing a new `json_data`, but unless I'm missing something, I don't think that variable is being used? If that's the case, consider not overwriting the `json_data` variable with the data from the new API call. Do you need to check the result of the `ensure_power_on` or `ensure_power_off` call before reporting back to the user?

Reply:
This should be in better shape now, I've created new variables for each type of power operation. As far as checking the return goes, I'm not sure it's completely necessary -- the `ensure_power_on/off` functions will either work (and return the json response data) or the module will fail. There may be edge cases, but, for the most part, I think my changes follow the spirit of what's already there. Once we get full integration tests going (very soon), it'll be less of a leap of faith.